src/samples/mqtt/bluemix.c

```diff
@@ -0,0 +1,355 @@
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief IBM Bluemix phonemotion demo client
+ *
+ * Sample client that connects IBM Bluemix phonemotion demo and simulates
+ * sensor data from a device.
+ *
+ * https://github.com/ibm-messaging/iotf-phonemotion
+ *
+ * To test, go to http://www.ibm.com/cloud-computing/bluemix/solutions/iot/
+ * and enter a user and a pin, then provide those credentials to the sample
+ *
+ * E.g: ./bluemix http://iotf.mybluemix.net/auth soletta_test 0000
+ */
+
+#include <math.h>
+#include <sol-http-client.h>
+#include <sol-json.h>
+#include <sol-log.h>
+#include <sol-mainloop.h>
+#include <sol-mqtt.h>
+
+struct bluemix_data {
+    /**
+     * HTML POST authentication
+     */
+    char *user;
+    char *pin;
+    char *url;
+
+    /**
+     * MQTT authentication
+     */
+    char *org_id;
+    char *client_id;
+    char *auth_token;
+
+    struct sol_timeout *connect_timeout;
+
+    /**
+     * MQTT Communication
+     */
+    struct sol_mqtt *mqtt;
+    struct sol_timeout *publish_timeout;
+};
+
+/**
+ * MQTT Data communication
+ */
+
+#define PI 3.14159265
+#define SENSOR_FMT "{\"d\": {\"ax\": %f, \"ay\": %f, \"az\": %f, \"oa\": %f, \"ob\": %f, \"og\": %f}}"
+
+static bool
+publish_callback(void *vdata)
+{
+    struct bluemix_data *data = (struct bluemix_data *)vdata;
+    struct sol_buffer payload = SOL_BUFFER_INIT_EMPTY;
+    struct sol_mqtt_message message;
+    char topic[] = "iot-2/evt/sensorData/fmt/json-iotf";
+    int r;
+    double v;
+    static int pulse = 0;
+
+    pulse = (pulse + 1) % 360;
+    v = sin(pulse * PI / 180) * 10;
+
+    r = sol_buffer_append_printf(&payload, SENSOR_FMT, v * 1.25, v * 1, v * 0.75, v * 20, v * 15, v * 10);
```

Comment:
so far, so good. But @ceolin is working on json serializer so we handle it properly. If users copy&paste this fragment on code that also calls `setlocale()`, then we may get float-points formatted with `,` instead of `.` for decimal and it will break the format.

sync with him so this is updated once those functions are merged.